From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LexManos <LexManos@gmail.com>
Date: Wed, 17 Jun 2020 11:07:09 -0700
Subject: [PATCH] Do not rebuild variable names in lambdas.

Breaks outer this references. Code existed before my time, no idea what it's intention is.

diff --git a/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java b/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
index 2a80835a255c98616352259a8c8a7c733c2ca19e..f9d15e108cfff05be4f38b10a00814259364a849 100644
--- a/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
+++ b/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
@@ -285,7 +285,7 @@ public class NestedClassProcessor {
     Set<String> setNewOuterNames = new HashSet<>(mapNewNames.values());
     setNewOuterNames.removeAll(method.setOuterVarNames);
 
-    method.varproc.refreshVarNames(new VarNamesCollector(setNewOuterNames));
+    //method.varproc.refreshVarNames(new VarNamesCollector(setNewOuterNames));
     method.setOuterVarNames.addAll(setNewOuterNames);
 
     for (Entry<VarVersionPair, String> entry : mapNewNames.entrySet()) {
